### PR TITLE
Fixes panel db query

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/3244

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [x] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
* clean profile
* visit some youtube publisher video
* make sure that publisher is recorded in ac table
* close browser
* open publisher db and modify reconcile stamp to something in the past
* open browser again and visit the same publisher
* make sure that correct publisher is displayed in the panel

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [x] Adequate test coverage exists to prevent regressions 
- [x] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: components/brave_rewards/browser/publisher_info_database.cc [hunk 1]
```diff
@@ -307,14 +307,17 @@ PublisherInfoDatabase::GetPanelPublisher(
...
+      "pi.provider, pi.verified, pi.excluded, "
+      "("
+      "SELECT IFNULL(percent, 0) FROM activity_info WHERE "
+      "publisher_id = ? AND reconcile_stamp = ? "
+      ") as percent "
+      "FROM publisher_info AS pi WHERE pi.publisher_id = ? LIMIT 1"));
 
   info_sql.BindString(0, filter.id);
   info_sql.BindInt64(1, filter.reconcile_stamp);
+  info_sql.BindString(2, filter.id);
```
Comment (ryanml) [inline components/brave_rewards/browser/publisher_info_database.cc]: Looks like has already been bound, is this unneeded?

Comment (jasonrsadler) [reply components/brave_rewards/browser/publisher_info_database.cc]: The args need to be bound, not the id. arg[2] hasn't been bound yet here.

Comment (NejcZdovc) [reply components/brave_rewards/browser/publisher_info_database.cc]: yup we need it as we have three `?` in the query and we use id in two places

File: components/brave_rewards/browser/publisher_info_database.cc [hunk 2]
```diff
@@ -307,14 +307,17 @@ PublisherInfoDatabase::GetPanelPublisher(
...
   sql::Statement info_sql(db_.GetUniqueStatement(
-      "SELECT pi.publisher_id, pi.name, pi.url, pi.favIcon, pi.provider, "
-      "pi.verified, pi.excluded, ai.percent FROM publisher_info AS pi "
-      "LEFT JOIN activity_info AS ai ON pi.publisher_id = ai.publisher_id "
-      "WHERE pi.publisher_id=? AND "
-      "(ai.reconcile_stamp = ? OR ai.percent IS NULL) LIMIT 1"));
+      "SELECT pi.publisher_id, pi.name, pi.url, pi.favIcon, "
+      "pi.provider, pi.verified, pi.excluded, "
+      "("
+      "SELECT IFNULL(percent, 0) FROM activity_info WHERE "
```
Comment (jasonrsadler) [inline components/brave_rewards/browser/publisher_info_database.cc]: What is the subquery for? Why are we filtering on activity table (percent, reconcileStamp) info if we're just pulling panel info? Shouldn't we just be able to pull it from publisher_info?

Comment (jasonrsadler) [reply components/brave_rewards/browser/publisher_info_database.cc]: ignore. I know the answer to that.

**Pull request merged**